prime-router/docs/design/proposals/mapping-inventory-test-plan.md

```diff
@@ -0,0 +1,50 @@
+# Mapping Inventory Test Plan
+
+## Problem Statement
+As a ReportStream engineer, I want tests that will validate the following statement: "The Universal Pipeline (UP) shall 
+be able to transform valid NIST 2.5.1 messages losslessly to FHIR using the official HL7v2 mapping inventory as a guide. 
+UP shall be able to transform FHIR ELR messages to valid NIST 2.5.1."
+
+Currently, we have integration tests that test an unknown combination of parts of sender transforms, receiver 
+transforms, and mapping inventory rules. We want a clear and definitive way of knowing what is already tested, and we 
+want the mapping inventory testing to be thorough. This means testing not only that each piece of the message ends up 
+in the Primary Target (as defined in column J of the mapping inventory spreadsheet), but also that the conditions set
+forth in the Computable ANTLR column of the mapping inventory spreedsheet are tested. This will just test moving from 
+HL7 V2 -> FHIR. We also need to test that the resulting FHIR message can be converted back to HL7 V2 losslessly. To 
+achieve this, we can reverse the order of the files and ensure that the output from processing the FHIR file matches 
+the HL7 V2 file. This will help keep the number of files needed down. 
+
+Integration tests are currently rather challenging to read. At a minimum, we need a standard for naming the files. When
+testing mapping inventory segments, we will name files with the format `segment_#`. An example, 
+`PID_1`. The corresponding FHIR file will be named the same, but will have a `.fhir` extension. This way, it is 
```

Comment:
One other thought I had was whether it would be worth having tests that go HL7 -> FHIR -> HL7.  I could see the argument that we wouldn't need those, so maybe it's fine to tackle that when a specific need arises.